client/client_test.go

```diff
@@ -337,21 +337,6 @@ func TestHTTPClusterClientDo(t *testing.T) {
 			wantPinned: 1,
 		},
 
-		// context.DeadlineExceeded short-circuits Do
```

Comment:
DeadlineExceeded cannot short-circuit Do anymore, and original code is not the correct way to test DeadlineExceeded return.